runtime/delegates_test.go

```diff
@@ -23,106 +22,102 @@ func TestDelegateRuntimeAPIs(t *testing.T) {
 		t.Parallel()
 
 		env, err := testutils.Setup()
-		if err != nil {
-			t.Skipf("Failed to setup test environment: %v", err)
-		}
+		require.NoError(t, err, "Failed to setup test environment")
 		defer env.Teardown()
 
 		blockHash, err := env.Client.Api.RPC.Chain.GetBlockHashLatest()
 		require.NoError(t, err, "Failed to get latest block hash")
 
 		delegates, err := runtime.GetDelegates(env.Client, &blockHash)
-
-		if err != nil {
-			assert.Contains(t, err.Error(), "no delegates found",
-				"Expected 'no delegates found' error in test environment, got: %v", err)
-		} else {
-			assert.NotNil(t, delegates, "Delegates should not be nil")
-			t.Logf("Found %d delegates in test environment", len(delegates))
-
-			if len(delegates) > 0 {
-				delegate := delegates[0]
-				assert.NotNil(t, delegate.AccountID, "Delegate AccountID should not be nil")
-				assert.GreaterOrEqual(t, uint64(delegate.TakeRate.Int64()), uint64(0), "TakeRate should be non-negative")
-				assert.GreaterOrEqual(t, uint64(delegate.NominatorStake.Int64()), uint64(0), "NominatorStake should be non-negative")
-				assert.GreaterOrEqual(t, uint64(delegate.ValidatorStake.Int64()), uint64(0), "ValidatorStake should be non-negative")
-				assert.GreaterOrEqual(t, uint64(delegate.TotalStake.Int64()), uint64(0), "TotalStake should be non-negative")
-				assert.NotNil(t, delegate.Registrations, "Registrations should not be nil")
-				assert.GreaterOrEqual(t, uint64(delegate.VotingPower.Int64()), uint64(0), "VotingPower should be non-negative")
-				assert.NotNil(t, delegate.ValidatorPermits, "ValidatorPermits should not be nil")
-				assert.GreaterOrEqual(t, uint64(delegate.Return.Int64()), uint64(0), "Return should be non-negative")
-
-				t.Logf("First delegate: %x", delegate.AccountID.ToBytes())
-				t.Logf("  TakeRate: %d", delegate.TakeRate.Int64())
-				t.Logf("  TotalStake: %d", delegate.TotalStake.Int64())
-				t.Logf("  VotingPower: %d", delegate.VotingPower.Int64())
-			}
+		require.NoError(t, err, "Failed to get delegates")
+		assert.NotNil(t, delegates, "Delegates should not be nil")
+
+		t.Logf("Found %d delegates in test environment", len(delegates))
+
+		switch len(delegates) {
```

Comment:
A test should be deterministic and do the same thing every time. You replaced an if statement with a different form of an if statement. If your test case doesnt add delegates before calling `getDelegates`, then your test case will never get to the default case. This does not make sense. The test case should cover both instances, when there are no delegates and where there are delegates.